src/App.tsx

```diff
@@ -1,24 +1,190 @@
 /* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 
-const USER_ID = 0;
+import { getTodos, deleteTodos, updateTodoOnServer } from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Filter } from './components/Filter';
+import { Error } from './components/Error';
+import { Todo, TodoStatus } from './types/Todo';
+import { ErrorMessage } from './types/ErrorMessage';
+import { filterTodos } from './components/FilterTodo';
+import { UserIdContext } from './utils/context';
+import { ChangeFunction } from './types/ChangeFunction';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState(TodoStatus.All);
+  const [errorType, setErrorType] = useState(ErrorMessage.None);
+  const [creatingTodoTitle, setCreatingTodoTitle] = useState('');
+  const [isClearCompleted, setIsClearCompleted] = useState(false);
+  const [isAllToggled, setIsAllToggled] = useState(false);
+
+  const userId = useContext(UserIdContext);
+
+  useEffect(() => {
+    getTodos(userId)
+      .then((userTodos) => setTodos(userTodos))
+      .catch(() => {
+        setErrorType(ErrorMessage.Download);
+      });
+  }, [userId]);
+
+  const counterActiveTodos = useMemo(
+    () => filterTodos(todos, TodoStatus.Active).length,
+    [todos],
+  );
+
+  const counterCompletedTodos = todos.length - counterActiveTodos;
+
+  const filteredTodos = useMemo(
+    () => filterTodos(todos, selectedFilter),
+    [selectedFilter, todos],
+  );
+
+  const showError = useCallback((error: ErrorMessage) => {
+    setErrorType(error);
+  }, []);
+
+  const hideError = useCallback(() => {
+    setErrorType(ErrorMessage.None);
+  }, []);
+
+  const AddTodo = useCallback(
```

Comment:
don't start variable name for common values from capital letter
```suggestion
  const addTodo = useCallback(
```